kernel/packages/decentraland-ecs/src/ecs/Engine.ts

```diff
@@ -264,7 +264,34 @@ export class Engine implements IEngine {
   }
 
   getComponentGroup(...requires: ComponentConstructor<any>[]) {
-    // TODO: memoize?
+
+    // Return an already created component-group if it already exists
+    if (requires.length > 0) {
+      // 1. get component groups for first require
+      let componentGroups = this._componentGroups[getComponentName(requires[0])]
+
+      if (componentGroups) {
+        const components = requires.slice()
+
+        // 2. search for a component group that has all the same requirements
+        componentGroups.forEach(componentGroup => {
+          if (components.length === componentGroup.requiresNames.length) {
+            for (let index = 0; index < components.length; index++) {
+              const componentName = getComponentName(components[index])
```

Comment:
Component class should be used instead of name (names can be repeated)

Reply:
With component class this doesn't work, it never finds a previously created component group, that was my first aproach.

Reply:
tried again after refactoring and using plain components works fine now